Realm/RLMRealm.h

```diff
@@ -91,6 +98,26 @@ NS_ASSUME_NONNULL_BEGIN
 + (instancetype)realmWithURL:(NSURL *)fileURL;
 
 /**
+ Asynchronously open a Realm and deliver it to a block on the given queue.
+
+ Opening a Realm asynchronously will perform all work needed to get the Realm to
+ a usable state (such as running potentially time-consuming migrations) on a
+ background thread before dispatching to the given queue. In addition,
+ synchronized Realms wait for all remote content available at the time the
+ operation began to be downloaded and available locally.
+
+ @param configuration A configuration object to use when opening the Realm.
+ @param callbackQueue The dispatch queue on which the callback should be run.
+ @param callback      A callback block. If the Realm was successfully opened,
+                      it will be passed in as an argument.
+                      Otherwise, an `NSError` describing what went wrong will be
+                      passed to the block instead.
+ */
++ (void)asyncOpenWithConfiguration:(RLMRealmConfiguration *)configuration
```

Comment:
The docs should mention that the Realm is still thread-confined and not queue-confined (and thus should not be held on to past the callback for anything but the main queue) as otherwise I can guarantee we'll eventually get a question from someone expecting it to be bound to that queue.